samza-api/src/main/java/org/apache/samza/startpoint/StartpointVisitor.java

```diff
@@ -18,54 +18,53 @@
  */
 package org.apache.samza.startpoint;
 
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.system.SystemAdmin;
 import org.apache.samza.system.SystemStreamPartition;
 
-
 /**
- * Visitor interface for system consumers to implement to support {@link Startpoint}s.
+ * A {@link SystemAdmin} implementation should implement this abstraction to support {@link Startpoint}'s.
  */
+@InterfaceStability.Evolving
 public interface StartpointVisitor {
 
   /**
-   * Seek to specific offset represented by {@link StartpointSpecific}
-   * @param systemStreamPartition The {@link SystemStreamPartition} to seek the offset to.
-   * @param startpointSpecific The {@link Startpoint} that represents the specific offset.
+   * Resolves the {@link StartpointSpecific} to a system specific offset.
```

Comment:
In the visitor pattern, the visitor interface is usually agnostic of what its implementations actually need to do. This keeps the visitor flexible in case we need it for something other than offset resolution in the future.
For example, another way to use the visitor could be for serializing `Startpoint`s into a store.